test/new-e2e/tests/apm/vm_test.go

```diff
@@ -0,0 +1,186 @@
+	s.Require().NoError(err)
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		testTraceAgentMetrics(s.T(), c, s.Env().FakeIntake)
+	}, 2*time.Minute, 10*time.Second, "Failed finding datadog.trace_agent.* metrics")
+}
+
+func (s *VMFakeintakeSuite) TestTracesHaveContainerTag() {
+	if s.transport != uds {
+		// TODO: Container tagging with cgroup v2 currently only works over UDS
+		// We should update this to run over TCP as well once that is implemented.
+		s.T().Skip("Container Tagging with Cgroup v2 only works on UDS")
+	}
+
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	s.Require().NoError(err)
+
+	service := fmt.Sprintf("tracegen-container-tag-%s", s.transport)
+
+	// Wait for agent to be live
+	s.T().Log("Waiting for Trace Agent to be live.")
+	s.Require().NoError(waitRemotePort(s, 8126))
+
+	// Run Trace Generator
+	s.T().Log("Starting Trace Generator.")
+	shutdown := runTracegenDocker(s.Env().RemoteHost, service, tracegenCfg{transport: s.transport})
+	defer shutdown()
+
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		testTracesHaveContainerTag(s.T(), c, service, s.Env().FakeIntake)
+	}, 2*time.Minute, 10*time.Second, "Failed finding traces with container tags")
+}
+
+func (s *VMFakeintakeSuite) TestStatsForService() {
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	s.Require().NoError(err)
+
+	service := fmt.Sprintf("tracegen-stats-%s", s.transport)
+
+	// Wait for agent to be live
+	s.T().Log("Waiting for Trace Agent to be live.")
+	s.Require().NoError(waitRemotePort(s, 8126))
+
+	// Run Trace Generator
+	s.T().Log("Starting Trace Generator.")
+	shutdown := runTracegenDocker(s.Env().RemoteHost, service, tracegenCfg{transport: s.transport})
+	defer shutdown()
+
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		testStatsForService(s.T(), c, service, s.Env().FakeIntake)
+	}, 2*time.Minute, 10*time.Second, "Failed finding stats")
+}
+
+func (s *VMFakeintakeSuite) TestBasicTrace() {
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	s.Require().NoError(err)
+
+	service := fmt.Sprintf("tracegen-basic-trace-%s", s.transport)
+
+	// Wait for agent to be live
+	s.T().Log("Waiting for Trace Agent to be live.")
+	s.Require().NoError(waitRemotePort(s, 8126))
+
+	// Run Trace Generator
+	s.T().Log("Starting Trace Generator.")
+	shutdown := runTracegenDocker(s.Env().RemoteHost, service, tracegenCfg{transport: s.transport})
+	defer shutdown()
+
+	s.T().Log("Waiting for traces.")
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		testBasicTraces(c, service, s.Env().FakeIntake, s.Env().Agent.Client)
+	}, 2*time.Minute, 10*time.Second, "Failed to find traces with basic properties")
+}
+
+func waitRemotePort(v *VMFakeintakeSuite, port uint16) error {
+	var (
+		c   net.Conn
+		err error
+	)
+	v.Eventually(func() bool {
+		v.T().Logf("Waiting for remote:%v", port)
+		// TODO: Use the e2e context
+		c, err = v.Env().RemoteHost.DialRemotePort(context.Background(), port)
```

Comment:
@vboulineau Not sure how to get the e2e.Context you mentioned. I don't see where I would get access to one anywhere, unless I need to hang on to a reference from the `Init` function.

Reply:
You'd need to add `context.Context` to `e2e.Context` - currently `e2e.Context` defines only a `T()` function to return a `*testing.T` instance. 

The suite initialiser needs some changes too, as we need to create a context, probably at `suite.init()`. Then the function can be `RemoteHost.DialRemotePort(port)`.

We can help with the implementation, but it looks like a bigger change than expected, so I suggest keeping your implementation and refactor it in a follow up, or wait for us to implement the change, or again open a new PR that introduces only the `remoteHost.DialPort` function.